idea/data-storage/src/gear/gear-event-listener.ts

```diff
@@ -276,30 +257,57 @@ export class GearEventListener {
             data: { id, change },
           } = event.event as CodeChanged;
           const codeStatus = change.isActive ? CodeStatus.ACTIVE : change.isInactive ? CodeStatus.INACTIVE : null;
-
-          codes.push({
+          const updateCodeInput = {
             id: id.toHex(),
             genesis: this.genesis,
             status: codeStatus,
             timestamp,
             blockHash: block.createdAtHash.toHex(),
             expiration: change.isActive ? change.asActive.expiration.toString() : null,
             uploadedBy: tx.signer.inner.toHex(),
-          });
+            meta: null,
+          };
+
+          const metaHash = await getMetaHash(this.api.code, id.toHex());
+
+          if(metaHash) {
+            const meta = await this.metaService.getByHash(metaHash);
```

Comment:
Perhaps it would better to make method `getaOrCreate` that returns a Meta object anyway?